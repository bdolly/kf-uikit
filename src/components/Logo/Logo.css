.Logo {
    @apply
        relative
        h-full;
}

.Logo--img {
    @apply
        max-w-sm
        max-h-full
        inline-block;
}

.Logo--text {
}
