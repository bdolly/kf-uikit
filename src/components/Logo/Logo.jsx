import React from 'react';
import propTypes from 'prop-types';
import classes from 'classnames';
import logoFull from './logo_full.svg';
import logo from './logo.svg';

/**
 * The Kids first logo.
 * May be `full` with `Kids First Data Resource` titling or
 * `logo mark` with only the art asset.
 */
const Logo = ({kind, className}) => {

  const logoAsset = kind === 'logo mark' ? logo : logoFull;
  const logoClasses = classes(
      className,
      'Logo--img',
  )
  const alt = 'Kids First Data Resource Center logo';

  return (
    <div className='Logo'>
      <img
        className={logoClasses}
        alt={alt}
        src={logoAsset}
      />
      <h1 className='Logo--text text-screen-reader'>{alt}</h1>
    </div>
  )
}

Logo.propTypes = {
  /** The type of logo */
  kind: propTypes.oneOf(['full', 'logo mark']),
  /** Any additional classes to be applied to the logo */
  className: propTypes.string,
};

Logo.defaultProps = {
  kind: 'full',
  className: null,
};

export default Logo;
