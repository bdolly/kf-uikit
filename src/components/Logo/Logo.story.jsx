/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { select } from '@storybook/addon-knobs';
import Logo from './Logo';

const stories = storiesOf('Branding', module);

stories.add('Logo', () => (
  <Logo
    kind={select('Kind', ['logo mark', 'full'], 'full')}
  />
  ),
  {
    info: {
      text: `Fluid logo that scales up to a width of the small viewport 
        breakpoint, or the full height of it container.`
    }
  }
)
