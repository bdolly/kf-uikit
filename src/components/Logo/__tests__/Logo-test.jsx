import React from 'react';
import renderer from 'react-test-renderer';
import Logo from '../Logo';

['full', 'logo mark'].map((kind) => {
  ['small', 'medium', 'large'].map((size) => {
    it(`${size} ${kind} logo renders correctly`, () => {
        const tree = renderer.create(
                <Logo
                  kind={kind}
                  size={size}
                />
                ).toJSON();
          expect(tree).toMatchSnapshot();
    });
  });
});
