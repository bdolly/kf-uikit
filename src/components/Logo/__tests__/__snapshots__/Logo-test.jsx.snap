// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`large full logo renders correctly 1`] = `
<div
  className="Logo"
>
  <img
    alt="Kids First Data Resource Center logo"
    className="Logo--img"
    src="test-file-stub"
  />
  <h1
    className="Logo--text text-screen-reader"
  >
    Kids First Data Resource Center logo
  </h1>
</div>
`;

exports[`large logo mark logo renders correctly 1`] = `
<div
  className="Logo"
>
  <img
    alt="Kids First Data Resource Center logo"
    className="Logo--img"
    src="test-file-stub"
  />
  <h1
    className="Logo--text text-screen-reader"
  >
    Kids First Data Resource Center logo
  </h1>
</div>
`;

exports[`medium full logo renders correctly 1`] = `
<div
  className="Logo"
>
  <img
    alt="Kids First Data Resource Center logo"
    className="Logo--img"
    src="test-file-stub"
  />
  <h1
    className="Logo--text text-screen-reader"
  >
    Kids First Data Resource Center logo
  </h1>
</div>
`;

exports[`medium logo mark logo renders correctly 1`] = `
<div
  className="Logo"
>
  <img
    alt="Kids First Data Resource Center logo"
    className="Logo--img"
    src="test-file-stub"
  />
  <h1
    className="Logo--text text-screen-reader"
  >
    Kids First Data Resource Center logo
  </h1>
</div>
`;

exports[`small full logo renders correctly 1`] = `
<div
  className="Logo"
>
  <img
    alt="Kids First Data Resource Center logo"
    className="Logo--img"
    src="test-file-stub"
  />
  <h1
    className="Logo--text text-screen-reader"
  >
    Kids First Data Resource Center logo
  </h1>
</div>
`;

exports[`small logo mark logo renders correctly 1`] = `
<div
  className="Logo"
>
  <img
    alt="Kids First Data Resource Center logo"
    className="Logo--img"
    src="test-file-stub"
  />
  <h1
    className="Logo--text text-screen-reader"
  >
    Kids First Data Resource Center logo
  </h1>
</div>
`;
